src/python/gem5/simulate/simulator.py

```diff
@@ -105,6 +106,8 @@ def __init__(
         ] = None,
         expected_execution_order: Optional[List[ExitEvent]] = None,
         checkpoint_path: Optional[Path] = None,
+        max_ticks: Optional[int] = m5.MaxTick,
```

Comment:
I think this warrants further discussion because I don't fully understand it. Setting of the max ticks has been part of the simulator for some time so not exactly sure what you're asking to be changed here.

Reply:
Ok, I've come around to this. But, please see below with my comment on the `run` function.